docs/RFCS/schema_gossip.md

```diff
@@ -0,0 +1,40 @@
+- Feature Name: schema_distribution
+- Status: unstarted
+- Start Date: TBD
+- RFC PR: #1743
+- Cockroach Issue:
+
+# Summary
+
+This RFC suggests implementing eventually-consistent replication of the SQL schema to all Cockroach nodes in a cluster using gossip.
+
+# Motivation
+
+In order to support performant SQL queries, each gateway node must be able to address the data requested by the query. Today this requires the node to read `TableDescriptor`s from the KV map, which we believe (though we haven't measured) will cause a substantial performance hit which we can mitigate by actively gossiping the necessary metadata.
+
+# Detailed design
+
+The entire `keys.SystemDBSpan` span will have its writes bifurcated into gossip (with a TTL TBD) in the same way that `storage.(*Replica).maybeGossipConfigs()` works today. The gossip system will provide atomicity in propagating these modifications through the usual sequence number mechanism.
+
+Complete propagation of new metadata will take at most `numHops * gossipInterval` where `numHops` is the maximum number of hops between any node and the publishing node, and `gossipInterval` is the maximum interval between sequential writes to the gossip network on a given node.
+
+On the read side, metadata reads' behaviour will change such that they will read from gossip rather than the KV store. This will require plumbing a closure or a reference to the `Gossip` instance down to the `sql.Planner` instance.
```

Comment:
not all of them. We definitely want anything that will mutate the metadata to first do a kv read.
Obviously high-qps requests (select, insert, delete, update) will use the cache.
Everything else (show *) could do either. If they use the gossiped version, they'll be consistent with high-qps ops (meaning you can tell when your changes have been propagated). If they don't, they'll be consistent with metadata mutators (meaning sequences of `GRANT|REVOKE ...` and `SHOW GRANTS` would be easier).